src/tests/ftest/util/server_utils.py

```diff
@@ -50,6 +57,439 @@ class ServerFailed(Exception):
+
+            If no key matches are found in the yaml file the BasicParameter
+            object will be set to its default value.
+
+            Args:
+                test (Test): avocado Test object
+            """
+            super(DaosServer.DaosServerConfig, self). get_params(test)
+            # Read the baseline conf file data/daos_server_baseline.yml
+            try:
+                with open('{}/{}'.format(test.basepath, DEFAULT_FILE), 'r') \
+                as rfile:
+                    self.data = yaml.safe_load(rfile)
+            except Exception as err:
+                self.log.info("<SERVER> Exception occurred: %s", str(err))
+                traceback.print_exception(
+                    err.__class__, err, sys.exc_info()[2])
+                raise ServerFailed(
+                    "Failed to Read {}/{}".format(test.basepath, DEFAULT_FILE))
+
+            # Read the values from avocado_testcase.yaml file if test ran
+            # with Avocado.
+            new_value_set = {}
+            if "AVOCADO_TEST_DATADIR" in os.environ:
+                avocado_yaml_file = str(
+                    os.environ["AVOCADO_TEST_DATADIR"]).split(".")[0] + ".yaml"
+
+                # Read avocado test yaml file.
+                try:
+                    with open(avocado_yaml_file, 'r') as rfile:
+                        filedata = rfile.read()
+                    # Remove !mux for yaml load
+                    new_value_set = yaml.safe_load(
+                        filedata.replace('!mux', ''))
+                except Exception as err:
+                    self.log.info("<SERVER> Exception occurred: %s", str(err))
+                    traceback.print_exception(
+                        err.__class__, err, sys.exc_info()[2])
+                    raise ServerFailed(
+                        "Failed to Read {}".format(
+                            '{}.tmp'.format(avocado_yaml_file)))
+
+            # Update values from avocado_testcase.yaml in DAOS yaml variables.
+            if new_value_set:
+                if 'server' in new_value_set['server_config']:
+                    for key in new_value_set['server_config']['server']:
+                        self.data['servers'][0][key] = \
+                            new_value_set['server_config']['server'][key]
+                for key in new_value_set['server_config']:
+                    if 'server' not in key:
+                        self.data[key] = \
+                            new_value_set['server_config'][key]
+
+            # Check if nvme and scm are enabled
+            srv_cfg = self.data['servers'][0]
+            if 'bdev_class' in srv_cfg and srv_cfg['bdev_class'] == "nvme":
+                self.nvme = True
+            if 'scm_class' in srv_cfg and srv_cfg['scm_class'] == "dcpm":
+                self.scm = True
+
+            # if specific log file name specified use that
+            if test.server_log:
```

Comment:
If `TestWithoutServers` or `Test` classes are passed into this method, it will need to beefed up a bit:
```suggestion
            if hasattr(test, "server_log") and test.server_log is not None:
```